File: in_Java/0443 String Compression.java
class Solution {
    public int compress(char[] chars) {
        int id = 0;
        int curlen;
        int i = 0, j = 0;
        while(j < chars.length){
            while(j < chars.length && chars[i] == chars[j]) j++;
            curlen = j - i;
            if(curlen == 1){
                chars[id] = chars[i];
                id++;
            }
            else{
                chars[id] = chars[i];
                char[] t = String.valueOf(curlen).toCharArray();
                int L = 0;
                while(L < t.length) chars[++id] = t[L++];
                id++;
            }
            i = j;
        }
        return id;
    }
}

File: in_Python/0443 String Compression.py
class Solution(object):
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        i = 0
        idx = 0
        j = 0
        N = len(chars)
        while j < N:
            j = i + 1
            while j < N and chars[j] == chars[i]:
                j += 1
            d = j - i
            if d == 1:
                chars[idx] = chars[i]
                idx += 1
            else:
                chars[idx] = chars[i]
                idx += 1
                if d < 10:
                    chars[idx] = str(d)
                    idx += 1
                else:
                    d = list(str(d))
                    for x in d:
                        chars[idx] = x
                        idx += 1
            i = j
        return idx